### PR TITLE
Remove almost everything using dust-tt/types in the extension

### PR DESCRIPTION
**Pull request by Fraggle**
## Description

On the path of removing dust-tt/types from the extensions.
(only `compareAgentForSort()` remains, as I want to discuss with @PopDaph & @tdraier first)

## Risk

None

## Deploy Plan

Deploy `front`

### CODE REVIEW

File: extension/app/src/components/input_bar/InputBar.tsx [hunk 1]
```diff
@@ -19,23 +18,6 @@ import { useContext, useEffect, useMemo, useRef, useState } from "react";
 
 // AGENT MENTION
 
-function AgentMention({
```
Comment (tdraier) [inline extension/app/src/components/input_bar/InputBar.tsx:22]: nit: remove comment ⬆️ 
we also have the same unused code in front/InputBar, can you remove it there too ?

**Review by tdraier — APPROVED** [review]
LGTM 👍

**Review by PopDaph — APPROVED** [review]
Thanks! 🙏🏻

File: sdks/js/src/types.ts [hunk 2]
```diff
@@ -124,6 +124,21 @@ const SupportedContentFragmentTypeSchema = FlexibleEnumSchema([
   ...(Object.keys(supportedLegacy) as [keyof typeof supportedLegacy]),
 ]);
 
+const uniq = <T>(arr: T[]): T[] => Array.from(new Set(arr));
```
Comment (flvndvd) [inline sdks/js/src/types.ts:127]: `nit` you can use lodash `uniqBy`.

**Review by flvndvd — APPROVED** [review]

File: extension/app/src/lib/utils.ts [hunk 3]
```diff
@@ -152,3 +152,44 @@ export function filterAndSortAgents(
...
+  // Check for 'dust'
+  if (a.sId === "dust") {
+    return -1;
+  }
+  if (b.sId === "dust") {
+    return 1;
+  }
+
+  // Check for 'gpt4'
+  if (a.sId === "gpt-4") {
```
[the remark this answers is not in the document]
Comment (Fraggle) [reply extension/app/src/lib/utils.ts:178]: sure

File: sdks/js/src/types.ts (hunk 2 above)
Comment (Fraggle) [reply sdks/js/src/types.ts:127]: I don't want to add a dependancy just for that.

File: extension/app/src/lib/auth.ts [hunk 4]
```diff
@@ -85,5 +88,5 @@ const fetchMe = async (token: string): Promise<UserTypeWithWorkspaces> => {
     );
   }
   const me = await response.json();
-  return me.user;
```
Comment (PopDaph) [inline extension/app/src/lib/auth.ts:88]: Just for info, we did not caught it during review, this broke the extension which is still calling /me with a fetch 🙈 
(fixed now)

**Pull request merged**